src/query/api/v1/handler/graphite/find_parser.go

```diff
@@ -29,16 +29,40 @@ import (
 	"github.com/m3db/m3/src/query/errors"
 	"github.com/m3db/m3/src/query/graphite/graphite"
 	graphiteStorage "github.com/m3db/m3/src/query/graphite/storage"
+	"github.com/m3db/m3/src/query/models"
 	"github.com/m3db/m3/src/query/storage"
 	"github.com/m3db/m3/src/query/util/json"
 	"github.com/m3db/m3/src/x/net/http"
 )
 
-func parseFindParamsToQuery(r *http.Request) (
-	*storage.FetchQuery,
+// parseFindParamsToQueries parses an incoming request to two find queries,
+// which are then combined to give the final result.
+// It returns, in order:
+// terminatedQuery, which adds an explicit terminator after the last term in
+//  the given query; this will return all values for exactly that tag which have
+//  no child nodes
+// childQuery, which adds an explicit match all after the last term in the given
+//  query; this will return all values for exactly that tag which have at least
+//  one child node.
+// query string, which is the initial query request (bar final matcher), which
+//  is used to reconstruct the return values.
+//  any error encountered during parsing.
+//
+// As an example, given the query `a.b*`, and metrics `a.bar.c` and `a.biz`,
+// terminatedQuery will return only [biz], and childQuery will return only
+// [bar].
+func parseFindParamsToQueries(r *http.Request) (
+	*storage.CompleteTagsQuery,
+	*storage.CompleteTagsQuery,
+	string,
```

Comment:
nit: what is this string supposed to be? 

i've taken to giving returned variables name for this reason in the method spec (but no naked returns in the code itself). i.e. something like:

```
func parseFindParamsToQueries(r *http.Request) (
terminatedQuery *storage.CompleteTagsQuery,
childQuery *storage.CompleteTagsQuery,
? string,
err *xhttp.ParseError,
) { 
values := r.URL.Query()
	query := values.Get("query")
	if query == "" {
		return nil, nil, "", xhttp.NewParseError(errors.ErrNoQueryFound, http.StatusBadRequest)
		// notice how this isn't just "return"
	}
```

Reply:
sgtm